stub/src/main/java/io/grpc/stub/ClientCalls.java

```diff
@@ -482,7 +485,7 @@ public void onClose(Status status, Metadata trailers) {
         if (status.isOk()) {
           buffer.add(BlockingResponseStream.this);
         } else {
-          buffer.add(status);
+          buffer.add(status.asRuntimeException(trailers));
```

Comment:
Using an exception to pass the trailers is pretty heavy-weight. Could you swap to something that doesn't have to create a back trace?